### ISSUE TITLE
Duplicate paths in sciond

### ISSUE THREAD

**Issue by FR4NK-W** [scionproto/scion#2444]
Path filtering issue in path combinator:
Multiple identical paths (same interfaces) are returned by sciond.

How to reproduce:
Used topology: modified Tiny.topo, add an intermediate AS with redundant links.

```
diff --git a/topology/Tiny.topo b/topology/Tiny.topo
index 8b564b1..498959c 100644
--- a/topology/Tiny.topo
+++ b/topology/Tiny.topo
@@ -11,9 +11,13 @@ ASes:
     cert_issuer: 1-ff00:0:110
   "1-ff00:0:112": # old 1-13
     cert_issuer: 1-ff00:0:110
+  "1-ff00:0:113":
+    cert_issuer: 1-ff00:0:110
 links:
   - {a: "1-ff00:0:110#1", b: "1-ff00:0:111#41", linkAtoB: CHILD, mtu: 1280}
-  - {a: "1-ff00:0:110#2", b: "1-ff00:0:112#1", linkAtoB: CHILD, bw: 500}
+  - {a: "1-ff00:0:110#2", b: "1-ff00:0:111#42", linkAtoB: CHILD, mtu: 1280}
+  - {a: "1-ff00:0:111#2", b: "1-ff00:0:112#1", linkAtoB: CHILD, bw: 500}
+  - {a: "1-ff00:0:111#3", b: "1-ff00:0:113#1", linkAtoB: CHILD, bw: 500}
 CAs:
   CA1-1:
     ISD: 1
```

showpaths output:
`ubuntu@ubuntu-xenial:~/go/src/github.com/scionproto/scion$ $SC/bin/showpaths -srcIA 1-ff00:0:112 -sciond /run/shm/sciond/sd1-ff00_0_112.sock -dstIA 1-ff00:0:113
Available paths to 1-ff00:0:113
[ 0] Hops: [1-ff00:0:112 1>2 1-ff00:0:111 3>1 1-ff00:0:113] Mtu: 1472
[ 1] Hops: [1-ff00:0:112 1>2 1-ff00:0:111 3>1 1-ff00:0:113] Mtu: 1472
[ 2] Hops: [1-ff00:0:112 1>2 1-ff00:0:111 3>1 1-ff00:0:113] Mtu: 1472
[ 3] Hops: [1-ff00:0:112 1>2 1-ff00:0:111 3>1 1-ff00:0:113] Mtu: 1472
`

expected showpaths output:
`ubuntu@ubuntu-xenial:~/go/src/github.com/scionproto/scion$ $SC/bin/showpaths -srcIA 1-ff00:0:112 -sciond /run/shm/sciond/sd1-ff00_0_112.sock -dstIA 1-ff00:0:113
Available paths to 1-ff00:0:113
[ 0] Hops: [1-ff00:0:112 1>2 1-ff00:0:111 3>1 1-ff00:0:113] Mtu: 1472
`

**Comment by FR4NK-W** [comment]
Proposed fix in PR #2445
could also be moved to FilterLongPaths, filtering all useless paths

**Comment by scrye** [comment]
The definition of duplicate paths has changed slightly following the new data-plane path format, but https://github.com/scionproto/scion/pull/3930 should now eliminate duplicates. Closing this, but feel free to reopen if there are still some duplicates causing issues.
